### ISSUE TITLE
Deprecation warnings in Rails 7.1.0.beta1

### ISSUE THREAD

**Issue by jclusso** [elastic/apm-agent-ruby#1420]
## Describe the bug

Deprecation warnings in Rails 7.1.0.beta1.

```
DEPRECATION WARNING: Calling deprecate_methods on ActiveSupport::Deprecation is deprecated and will be removed from Rails (use your own Deprecation object instead) (called from require at /Users/jarrett/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/bundler/gems/apm-agent-ruby-c62fbc5be97e/lib/elastic_apm/spies.rb:111)
DEPRECATION WARNING: Calling deprecate_methods on ActiveSupport::Deprecation is deprecated and will be removed from Rails (use your own Deprecation object instead) (called from require at /Users/jarrett/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/bundler/gems/apm-agent-ruby-c62fbc5be97e/lib/elastic_apm/spies.rb:111)
DEPRECATION WARNING: Calling deprecate_methods on ActiveSupport::Deprecation is deprecated and will be removed from Rails (use your own Deprecation object instead) (called from require at /Users/jarrett/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/bundler/gems/apm-agent-ruby-c62fbc5be97e/lib/elastic_apm/spies.rb:111)
DEPRECATION WARNING: Calling deprecate_methods on ActiveSupport::Deprecation is deprecated and will be removed from Rails (use your own Deprecation object instead) (called from require at /Users/jarrett/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/bundler/gems/apm-agent-ruby-c62fbc5be97e/lib/elastic_apm/spies.rb:111)
```

## Steps to reproduce
1. Upgrade to Rails 7.1.0.beta1
2. Run any rails command like `rails server` or `rails console`

## Expected behavior

No deprecation warnings.

## Environment
- **Ruby version**: 3.2.2
- **Framework and version**: Rails 7.1.0.beta1
- **Agent version**: 4.7.0

## Additional context

I tried to figure out how this worked to open a PR. If anyone can provide some direction I'll investigate this further myself.

**Comment by picandocodigo** [comment]
Hi @jclusso,
I tried reproducing the error using a new Rails project with rails `7.1.0.beta1`, Ruby 3.2.2 and the agent version `4.7.0`, but I couldn't find the same deprecation warning.

The line the error is referring to in `spies.rb` is when a library is required, so the code is coming from a required library, not from code in the agent. I looked into the Rails code and found [this Pull Request](https://github.com/rails/rails/pull/47354/) which deprecates the singleton usage of `ActiveSupport::Deprecation`. So there needs to be an instance of ActiveSupport::Deprecation calling `deprecate_methods`. I couldn't find a singleton usage of Deprecation in the Rails code, is it possible that some of your Rails application code is using `ActiveSupport::Deprecation.deprecate_methods`?

Have you updated your code to Rails 7.1.1 and if so, is this still happening?
Thanks!
